metals/src/main/scala/scala/meta/internal/metals/ReferenceProvider.scala

```diff
@@ -98,13 +99,142 @@ final class ReferenceProvider(
     }
   }
 
+  private def toReferenceParams(
+      params: TextDocumentPositionParams,
+      includeDeclaration: Boolean
+  ): ReferenceParams = {
+    toReferenceParams(
+      params.getTextDocument,
+      params.getPosition,
+      includeDeclaration
+    )
+  }
+
+  private def toReferenceParams(
+      textDoc: TextDocumentIdentifier,
+      pos: Position,
+      includeDeclaration: Boolean
+  ): ReferenceParams = {
+    val referenceParams = new ReferenceParams()
+    referenceParams.setPosition(pos)
+    referenceParams.setTextDocument(textDoc)
+    val context = new ReferenceContext()
+    context.setIncludeDeclaration(includeDeclaration)
+    referenceParams.setContext(context)
+    referenceParams
+  }
+
+  def toReferenceParams(
+      location: Location,
+      includeDeclaration: Boolean
+  ): ReferenceParams = {
+    val textDoc = new TextDocumentIdentifier()
+    textDoc.setUri(location.getUri)
+    toReferenceParams(
+      textDoc,
+      location.getRange.getStart,
+      includeDeclaration
+    )
+  }
+
+  private def implementations(
+      textParams: TextDocumentPositionParams,
+      shouldCheckImplementation: Boolean
+  ): Seq[Location] = {
+    if (shouldCheckImplementation) {
+      for {
+        implLoc <- implementation.implementations(textParams)
+        locParams = toReferenceParams(implLoc, includeDeclaration = true)
+        loc <- references(locParams).locations
+      } yield loc
+    } else {
+      Nil
+    }
+  }
+
+  def allRefs(
+      symbolOccurrence: SymbolOccurrence,
+      doc: TextDocument,
+      fnIncludeSynthetics: Synthetic => Boolean,
+      textParams: TextDocumentPositionParams
+  ): Seq[Location] = {
+    def toTextParams(location: Location): TextDocumentPositionParams = {
```

Comment:
I think yes, I just moved this part of code from RenameProvider as it was.
If I move it to outside of this method then `parentSymbols.map(toTextParams)` below will have to change because `toTextParams` will need to get additional params. Move it anyway?